### PR TITLE
8.0.2 Recording, Timer and EPG Improvements

### PR DESCRIPTION
**Pull request by emveepee**
Limit timeline on in progress recordings.   Add logic for -1 lastplayedposition = watched condition in Kodi.

### CODE REVIEW

**Comment by emveepee** [comment]
@AlwinEsch I don't see this PR in jenkins  https://jenkins.kodi.tv/view/Matrix%20Binary%20Add-ons/job/kodi-pvr/job/pvr.nextpvr/view/change-requests/   Do I need to change something on the PR?  I use that for testing Android artifacts.

**Comment by phunkyfish** [comment]
There appears to be a problem with jenkins on PVR projects currently. Will look into it. FYI, jenkins does run on releases. It's only on CI it is missing.

**Review by phunkyfish — APPROVED** [review]
LGTM

File: src/Timers.cpp [hunk 1]
```diff
@@ -593,6 +593,15 @@ PVR_ERROR Timers::AddTimer(const kodi::addon::PVRTimer& timer)
   const int epgOid = m_epgOidLookup[oidKey];
   kodi::Log(ADDON_LOG_DEBUG, "TIMER_%d %s", epgOid, oidKey.c_str());
   std::string request;
+
+  int marginStart = timer.GetMarginStart();
+  int marginEnd = timer.GetMarginEnd();
+  if (m_settings.m_ignorePadding && timer.GetClientIndex() == PVR_TIMER_NO_CLIENT_INDEX && marginStart == 0 && marginStart == 0)
+  {
+    marginStart = m_settings.m_defaultPrePadding;
+    marginEnd= m_settings.m_defaultPostPadding;
```
Comment (phunkyfish) [inline src/Timers.cpp]: `marginEnd = `

**Review by phunkyfish — APPROVED** [review]
Just one minor

File: src/Timers.cpp (hunk 1 above)
Comment (emveepee) [reply src/Timers.cpp]: Oops thanks, that should be it for Matrix , until the user bugs roll in.  Thanks for your continued support.

Comment (phunkyfish) [reply src/Timers.cpp]: Your welcome

**Pull request merged**